homeassistant/components/vicare/climate.py

```diff
@@ -0,0 +1,204 @@
+        if desired_temperature == PYVICARE_ERROR:
+            desired_temperature = None
+
+        self._target_temperature = desired_temperature
+
+        self._current_mode = self._api.getActiveMode()
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS_HEATING
+
+    @property
+    def name(self):
+        """Return the name of the climate device."""
+        return self._name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def hvac_mode(self):
+        """Return current hvac mode."""
+        return VICARE_TO_HA_HVAC_HEATING.get(self._current_mode)
+
+    def set_hvac_mode(self, hvac_mode):
+        """Set a new hvac mode on the ViCare API."""
+        vicare_mode = HA_TO_VICARE_HVAC_HEATING.get(hvac_mode)
+        if vicare_mode is None:
+            _LOGGER.error(
+                f"cannot set invalid vicare mode: {hvac_mode} / {vicare_mode}"
+            )
+        else:
+            _LOGGER.debug(f"setting hvac mode to {hvac_mode} / {vicare_mode}")
+            self._api.setMode(vicare_mode)
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available hvac modes."""
+        return list(HA_TO_VICARE_HVAC_HEATING)
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return VICARE_TEMP_HEATING_MIN
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return VICARE_TEMP_HEATING_MAX
+
+    @property
+    def precision(self):
+        """Return the precision of the system."""
+        return PRECISION_WHOLE
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperatures."""
+        temp = kwargs.get(ATTR_TEMPERATURE)
+        if temp is not None:
+            self._api.setProgramTemperature(
+                self._current_program, self._target_temperature
+            )
+
+    @property
+    def preset_mode(self):
+        """Return the current preset mode, e.g., home, away, temp."""
+        return VICARE_TO_HA_PRESET_HEATING.get(self._current_program)
+
+    @property
+    def preset_modes(self):
+        """Return the available preset mode."""
+        return list(VICARE_TO_HA_PRESET_HEATING)
+
+    def set_preset_mode(self, preset_mode):
+        """Set new preset mode and deactivate any existing programs."""
+        vicare_program = HA_TO_VICARE_PRESET_HEATING.get(preset_mode)
+        if vicare_program is None:
+            _LOGGER.error(
+                f"cannot set invalid vicare program: {preset_mode} / {vicare_program}"
```

Comment:
Please use `%s` and pass arguments after the message string to format the log message.

https://docs.python.org/3/library/logging.html#logging.Logger.debug